From 5cc2b16b4882b9bc8c77573983b003b71fdbb50e Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 23 Jun 2014 20:01:07 -0400
Subject: [scsi] qla2xxx: Issue abort command for outstanding commands during cleanup when only firmware is alive

Message-id: <1403553672-3844-2-git-send-email-cdupuis@redhat.com>
Patchwork-id: 84411
O-Subject: [PATCH RHEL 7.1 e-stor 1/6] qla2xxx: Issue abort command for outstanding commands during cleanup when only firmware is alive.
Bugzilla: 1090378
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

From: Giridhar Malavali <giridhar.malavali@qlogic.com>

Bug 1090378 (https://bugzilla.redhat.com/show_bug.cgi?id=1090378)

scsi commit id af13b700de669e80c215236769649a7393a5e356

Signed-off-by: Giridhar Malavali <giridhar.malavali@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_nx.c b/drivers/scsi/qla2xxx/qla_nx.c
index 11ce53d..b850138 100644
--- a/drivers/scsi/qla2xxx/qla_nx.c
+++ b/drivers/scsi/qla2xxx/qla_nx.c
@@ -3668,8 +3668,10 @@ qla82xx_chip_reset_cleanup(scsi_qla_host_t *vha)
    for (cnt = 1; cnt < req->num_outstanding_cmds; cnt++) {
     sp = req->outstanding_cmds[cnt];
     if (sp) {
-     if (!sp->u.scmd.ctx ||
-         (sp->flags & SRB_FCP_CMND_DMA_VALID)) {
+     if ((!sp->u.scmd.ctx ||
+         (sp->flags &
+      SRB_FCP_CMND_DMA_VALID)) &&
+      !ha->flags.isp82xx_fw_hung) {
       spin_unlock_irqrestore(
           &ha->hardware_lock, flags);
       if (ha->isp_ops->abort_command(sp)) {
-- 
1.7.1